### ISSUE TITLE
Whitepaper Bounty – Issue 1: Cosmos DEX

### ISSUE THREAD

**Issue by MartyMe** [cosmos/cosmos#55]
@jaekwon @ebuchman 

Building a decentralised exchange is obviously a very ambitious and challenging project. I think the whitepaper does a good job explaining the limitations of AXC transactions and the obvious problems with centralised exchanges in general. 

I am missing one thing however: A comparison between your approach and the one the Waves platform is taking with their attempt of building a (sort of) decentralised exchange with centralised matching and decentralised settlement. In the Cosmos Plan it says: "The Cosmos DEX provides fast centralised matching with distributed custody of funds", which sounds quite similar to me.
  
You mention the strong network effect in the exchange world. If Waves is successful at creating their DEX (which is scheduled to be ready for release in the coming weeks I believe), what will the Cosmos DEX offer that would incentive traders to change exchanges? As far as I understand both projects claim that they are going to offer all the benefits of a centralised exchange (especially in terms of speed and using trading bots).

In order to add a new currency to the Cosmos DEX each currency would need its own zone connected to the Hub. In order to compete with Poloniex in terms of coin variety you would currently need around 80 zones. What would be the timeframe for that? Some of those zones also might not be used much and therefore not be very profitable for validators. How can you make sure those zones are still supported when there is a plan to loosen regulations regarding [the validators choice which zones to support](https://cosmos.network/faq#what-is-the-maximum-number-of-validators-in-cosmos-what-about-nodes).

**Comment by ebuchman** [comment]
Thanks for this. I'm not familiar with the Waves model, will have to look - sounds similar. The benefit would be interoperability with other elements of the Cosmos network - but maybe that will include Waves one day.

> In order to add a new currency to the Cosmos DEX each currency would need its own zone connected to the Hub

Not necessarily. The [application framework](github.com/tendermint/basecoin) we're using for the Hub and other zones is natively multi-asset, so there should be ways for new tokens to be created without resorting to creating a new zone - for instance, proposals accepted through the governance mechanism might create new coins on the hub with arbitrary allocations. The hub itself is a multi-asset decentralized exchange without an order book.

**Comment by antoineherzog** [comment]
@ebuchman yes please look into Waves, i am curious to have your insight about their technology ;)

**Comment by gamarin2** [comment]
@ebuchman +1 would like your insight too. What will be the competitive advantages of Cosmos Dex over Waves Dex ?

**Comment by howard-peng** [comment]
Want to know more detail about those 2 DEX protocol & both difference !! 

thx!!

**Comment by howardpen9** [comment]
@gamarin2  +1 want to know!

**Comment by gamarin2** [comment]
So I have done better due diligence on waves and I can answer this question now.

The design of the 2 Dexes is pretty similar in that there is a centralised order book and decentralised settlement layer. 

The main differences lies in how you get your token to the blockchains where trades will be settled. Currently, Waves uses "Gateways" which are basically centralised relayers. I could find little info about how they work in detail, but it seems like funds are held on a multisig on the bridged chain and that's it. They state that they will be fully decentralised using Atomic Swaps soon, but swaps have limitations:

1- If you use swaps to perform the trade, then you get issues. Notably, it's hard to have deep order books and traders can get locked in a position for some time if the other party does not complete its part of the trade.
2- If you use swaps to get the tokens to the platform, you run into the problem that you need something to swap the token for. Problem is that in this scenario, you juste want to get your BTC (say), from your address on the BTC chain to your address on the DEX chain. To do the swap, you need to swap the token for something, but what would be this something?

Cosmos DEX on the other hand provide full decentralisation with IBC. IBC is different than Atomic Swaps in that it enables fully decentralised 2-ways transfers between chains. You don't need to exchange your token for another to perform the transfer. Also, you don't need to have a centralised entity to act as a middleman, as two blockchains connected with IBC communicate directly with each other in a decentralised way (read IBC spec for more https://github.com/cosmos/cosmos-sdk/tree/develop/docs/spec/ibc). The catch is that IBC works with chains that have finality (like Tendermint based chain), and Proof-Of-Work chains like Bitcoin do not have this property. 

We use proxy-chains to connect these non-finality chains to Cosmos and eventually the DEXes that live within Cosmos. These proxy-chains are called "Peg-Zones". Contrary to Gateways, they are fully decentralised blockchains. The downside is that you can't easily replicate a Peg-Zone from one blockchain to another. Each Peg-Zone has to be designed for the specific blockchain it tries to bridge. And it's not trivial. The first Peg-Zone that we will deploy will be the Ethereum Peg-Zone. Then we'll go for Bitcoin, but that's more difficult.

**Comment by divinationstar** [comment]
Is there another place where discussion of a DEX on Cosmos has progressed?
